src/micro/attributes.html

```diff
@@ -83,7 +83,7 @@
         // NOTE: never allow 'class' to be set in hostAttributes
         // since shimming classes would make it work
         // inconsisently under native SD
-        if (!this.hasAttribute(n) && (n !== 'class')) {
+        if (!this.hasAttribute(n) && (n !== 'class') && this._config[n] === undefined) {
```

Comment:
What if we add an overridable `_canApplyAttribute` which we can implement in the configure module. This will allow us to avoid an extra setAttribute for this case. 

As for the case-map question, it seems that we aren't currently doing this mapping for hostAttributes at all so perhaps we can just ignore that? wdyt?